### ISSUE TITLE
Add options to `terraform show` to have parity with `terraform plan` output

### ISSUE THREAD

**Issue by devonbleak** [hashicorp/terraform#21596]
### Current Terraform Version

```
$ terraform -version
Terraform v0.12.1
+ provider.aws v2.13.0
```

### Use-cases

I'd like to have a way for `terraform show` to display the additional information that we get from a `terraform plan` to do quick sanity checks on the plans without having to re-run them.  

Currently there are 3 major differences in `terraform plan` vs `terraform show` output:

1. The summary line (`Plan: X to add, Y to change, Z to destroy`) is only present in `terraform plan` output
2. `terraform plan` shows additional details about the resources it's changing and why
3. `terraform show` displays a `?` status for resources that the plan doesn't touch

### Attempted Solutions

Looked at `terraform show -help`.  This could be solvable by piping JSON output into a separate process to produce the desired output but it seems TF should already have the code available for this since it exists for `terraform plan`

### Proposal

Proposing the following additions to `terraform show` (and definitely open to shorter/better suggestions on the names of the flags):

1. `-summarize` would add the summary line display
2. `-verbose` would add the additional details regarding the changes to resources
3. `-changed-only` would suppress output for resources that aren't changing

And/or `-plan` to just completely mimic `terraform plan` output and save some typing.

**Comment by apparentlymart** [comment]
Hi @devonbleak! Thanks for requesting this.

We noticed shortly after the 0.12.0 release that the `terraform show` command didn't get all of the same updates `terraform plan` did for the new plan rendering, which is why its output seems pretty "degraded" compared to `terraform plan`: it's mostly based on the v0.11 implementation, with some partial shims.

We do intend to update it to use the same rendering codepath as `terraform plan` does, though we don't have plans to make it have any additional options compared to the normal plan output... we _might_ do that in a separate change later, but output parity is the first goal.

That it shows resources that aren't changing is a bug and it should be fixed as part of switching to the new codepath. That arises because the Terraform v0.12 plan format includes all of the resources but the unchanged ones are filtered out during display, whereas the old v0.11 format (which the `terraform show` command is currently expecting) had them pre-filtered.

**Comment by devonbleak** [comment]
Hi @apparentlymart - thanks for the speedy response!

Your existing plan/prioritization sounds good.  

The only thing I would add is that in 0.11 the `terraform show` output also did not include the summary line, so would just want to be sure it's either included by default or there's a new option to include it.  Thanks!

**Comment by apparentlymart** [comment]
This was addressed in #23292. The change should be included in the next minor release.

The output from `terraform show` with a saved plan will match the output produced by `terraform plan` when it originally saved that plan, including the summary information.
